### PR TITLE
Fix timezone problems in django admin widget

### PR DESCRIPTION
**Pull request by KonstantinSchubert**
tl;dr: I think it fixes the problem that #71 is trying to address, but in a simpler and more idiomatic way.

## What is the problem? 

My browser is in timezone GMT+2.  If I select the **15th of October** in the widget, the Javascript in the current implementation will create the following datetime object:  

`2017-10-15T00:00+02:00`

**This is midnight in my timezone, but in UTC, this is actually still 10pm on the 14th!** Now, somewhere on the way to the django database, the timezone information gets stripped away and what is stored  is the time in UTC, without saying that it is UTC: 

`2017-10-14T22:00`

Clearly, this is a bug because we are on the wrong date: **14th of October instead of 15th**.

As far as I can see convention of this library appears to be that we store the timezone-agnostic date by storing a timezone-agnostic datetime with the time set to midnight. This result is clearly violating this convention.

## How does this PR fix this?

This PR fixes it by making sure that the widget creates the following datetime object instead:

`2017-10-15T02:00+02:00`

**This is 2 am in my timezone, but in UTC, this is actually midnight**

So, when the timezone information gets stripped during form submission, the resulting timezone-agnostic datetime that ends up in the database is 

`2017-10-15:00:00`

The 15th is the date that the user clicked. Since we are operating timezone-agnostic, this is all we can ask for. Happy end :)

### CODE REVIEW

**Comment by KonstantinSchubert** [comment]
@douwevandermeij , @fruitschen this might be interesting for you

**Comment by audiolion** [comment]
> 2017-10-15T02:00+02:00

It seems like calling `setUTCHours()` and the like would make it so the time when you hit `today` is
> 2017-10-15T00:00+00:00

To clarify, this fixes the `admin` widget when you select `today` option?

**Comment by KonstantinSchubert** [comment]
> To clarify, this fixes the admin widget when you select today option?

It fixes it for all dates in the date picker, not only for today!

**Comment by KonstantinSchubert** [comment]
Any news on this?

**Comment by KonstantinSchubert** [comment]
Bump

**Comment by jakub-borusewicz** [comment]
bump. Any plans to merge this in the near future? This bug is quite annoying.

**Comment by dominicrodger** [comment]
Apologies for missing this @KonstantinSchubert, and thanks for the bump @jakub-borusewicz. I'll aim to get this merged this week. If for some reason I miss merging it, feel free to bump this thread or ping me on Twitter (handle is dominicrodger there too), and I'll get it done. If someone could either update the PR to resolve the merge conflicts I'd be grateful, otherwise I'll take a look at doing that myself.

**Comment by dominicrodger** [comment]
Actually, scratch that, I resolved the single merge conflict with GH's GUI, didn't know that was a thing 😄 . I'd like to verify myself that this change fixes the problem (and indeed that there is a problem - it's not bitten me personally) before I get this merged, but I doubt that'll take more than 20 minutes or so, so I should be able to find time before the end of the week.

**Comment by dominicrodger** [comment]
Having taken a look - I don't think the patch is correct. I think we should definitely strip out the time portion of dates that are submitted, but I was seeing some odd behaviour with this patch. Here's what I was doing:

- Set Django's timezone to UTC, with ``USE_TZ`` set to True
- Set my computer's timezone to UTC - 8.
- Made a tiny model with a ``RecurrenceField``. Clicked "Add Date".

Today is August 4th, both in my computer's timezone, and in UTC. The selected date has August 3rd set. If I revert the patch, I get August 4th.

I think it's because we're making the change too early - the date returned from ``date_today`` is fractionally after midnight on August 4th, so something later presumably converts it to my PC's local time, which would be 4pm on August 3rd.

I'm not sure fixing the original bug in JS is the way to go - wouldn't it be easier to just strip times out of what's saved in the database? I may be missing something here - this isn't an area I've ever had problems with, so I've not digged into this a whole lot (I'm probably among the least affected possible users here, since I'm based in London, and am rarely online at 1am!).
